src/static/riot/competitions/detail/_tabs.tag

```diff
@@ -280,23 +278,37 @@
                     })
                 })
             })
-            if (!_.isEmpty(self.competition.leaderboards)) {
-                self.selected_leaderboard_index = self.competition.leaderboards[0].id
-            }
-            self.selected_phase_index = _.get(_.find(self.competition.phases, {'status': 'Current'}), 'id')
+
             self.competition.is_admin = CODALAB.state.user.has_competition_admin_privileges(competition)
-            self.update()
+            self.selected_phase_index = _.get(_.find(self.competition.phases, {'status': 'Current'}), 'id')
+            if (self.selected_phase_index == null) {
+                self.selected_phase_index = _.get(_.find(self.competition.phases, {is_final_phase: true}), 'id')
+            }
 
             $('.phases-tab .accordion', self.root).accordion()
 
             $('.tabular.pages-menu.menu .item', self.root).tab()
 
-            _.forEach(competition.pages, (page, index) => {
+            // Need to run update() to build tags to render html in
+            self.update()
+            _.forEach(self.competition.pages, (page, index) => {
                 $(`#page_${index}`)[0].innerHTML = render_markdown(page.content)
             })
-            _.forEach(competition.phases, (phase, index) => {
+            _.forEach(self.competition.phases, (phase, index) => {
                 $(`#phase_${index}`)[0].innerHTML = render_markdown(phase.description)
             })
+
+            CODALAB.api.get_leaderboard_submissions(self.competition.id)
+                .done( data => {
+                    self.leaderboard_phases = data
+                    if (!_.isEmpty(self.leaderboard_phases)) {
+                        self.phase_selected(_.find(self.leaderboard_phases, {'id': self.selected_phase_index}))
+                    }
+                })
+                .fail(error => {
+                    toastr.error(error)
+                })
+
```

Comment:
This is using that method I thought we could delete. I think there is a smarter way to go about this since we should already have the selected phase index in `self.selected_phase_index`